torch/_prims_common/__init__.py

```diff
@@ -504,23 +505,34 @@ def compute_elementwise_output_logical_to_physical_perm(
     shape = tensors[0].shape
 
     def should_swap(idx_a, idx_b):
+        def gte(a, b):
+            # semantics for a >= b, assuming a != b, a >= 0, b >= 0
+            if _guard_semantics(a == 0):
+                return False
+            if _guard_semantics(b == 0):
+                return True
+
+            expr = (a.node.expr if isinstance(a, torch.SymInt) else a) // (b.node.expr if isinstance(b, torch.SymInt) else b)
+            if isinstance(expr, (int, sympy.Integer)):
```

Comment:
i see here it sounds like you want
try:
   return expr>=1
except DataDepenedentErrpr:
              return not isinstance(expr, sympy.floor)

Reply:
maybe guard_or_else apI where you can pass continuation lambda?
I needed that in several occasions